Content.Server/Animals/Components/EggLayerComponent.cs

```diff
@@ -9,44 +11,47 @@ namespace Content.Server.Animals.Components;
 ///     It also grants an action to players who are controlling these entities, allowing them to do it manually.
 /// </summary>
 
-[RegisterComponent]
+[RegisterComponent, Access(typeof(EggLayerSystem))]
 public sealed partial class EggLayerComponent : Component
 {
-    [DataField]
-    public EntProtoId EggLayAction = "ActionAnimalLayEgg";
+    /// <summary>
+    ///     The item that gets laid/spawned, retrieved from animal prototype.
+    /// </summary>
+    [DataField(required: true), ViewVariables(VVAccess.ReadWrite)]
```

Comment:
```suggestion
    [DataField(required: true)]
```
DataField automatically gives VVAccess.ReadWrite

This goes for a bunch of other DataFields in this PR too